docs/advanced/document-store/redis.md

```diff
@@ -0,0 +1,334 @@
+ embedding=[2. 2. 2.],   score=5.96046447754e-08
+ embedding=[4. 4. 4.],   score=5.96046447754e-08
+```
+
+Then you can use a different search indexing schema on current DocumentArray as follows:
+```python
+da2 = DocumentArray(
+    storage='redis',
+    config={'n_dim': n_dim, 'update_schema': True, 'distance': 'L2'},
+)
+
+results = da.find(np_query, limit=n_limit)
+
+print('\nEmbeddings Approximate Nearest Neighbours:\n')
+for embedding, score in zip(
+    results.embeddings,
+    results[:, 'scores'],
+):
+    print(f' embedding={embedding},\t score={score["score"].value}')
+```
+
+This would print:
+
+```text
+Embeddings Approximate Nearest Neighbours:
+
+ embedding=[8. 8. 8.],   score=0
+ embedding=[9. 9. 9.],   score=3
+ embedding=[7. 7. 7.],   score=3
+ embedding=[6. 6. 6.],   score=12
+ embedding=[5. 5. 5.],   score=27
+```
+
+
+## Config
+
+The following configs can be set:
+
+| Name              | Description                                                                                       | Default                                           |
+|-------------------|---------------------------------------------------------------------------------------------------|-------------------------------------------------- |
+| `host`            | Host address of the Redis server                                                                  | `'localhost'`                                     |
+| `port`            | Port of the Redis Server                                                                          | `6379`                                            |
+| `redis_config`    | Other Redis configs in a Dict and pass to `Redis` client constructor, e.g. `socket_timeout`, `ssl`| `{}`                                              |
+| `index_name`      | Redis index name; the name of RedisSearch index to set this DocumentArray                         | `'idx'`                                           |
+| `n_dim`           | Dimensionality of the embeddings                                                                  | `None`                                            |
+| `flush`           | Boolean flag indicating whether to clear previous DocumentArray in Redis                          | `False`                                           |
+| `update_schema`   | Boolean flag indicating whether to update Redis Search schema                                     | `True`                                            |
+| `distance`        | Similarity distance metric in Redis                                                               | `'COSINE'`                                        |
+| `batch_size`      | Batch size used to handle storage updates                                                         | `64`                                              |
+| `method`          | Vector similarity index algorithm in Redis                                                        | `'HNSW'`                                          |
```

Comment:
can we say `Vector similarity index algorithm in Redis` -> `Vector similarity index algorithm in Redis, either `FLAT` or `HNSW`  `